src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -14,18 +14,27 @@ namespace System.Reflection
     //
     // Parses an assembly name.
     //
-    [System.Runtime.CompilerServices.ReflectionBlocked]
-    public static class AssemblyNameParser
+    internal static class AssemblyNameParser
     {
-        public static void Parse(AssemblyName blank, string s)
+        public struct AssemblyNameParts
         {
-            if (s == null)
-                throw new ArgumentNullException(nameof(s));
-            RuntimeAssemblyName runtimeAssemblyName = Parse(s);
-            runtimeAssemblyName.CopyToAssemblyName(blank);
+            public AssemblyNameParts(string name, Version? version, string? cultureName, AssemblyNameFlags flags, byte[]? publicKeyOrToken)
+            {
+                m_name = name;
+                m_version = version;
+                m_cultureName = cultureName;
+                m_flags = flags;
+                m_publicKeyOrToken = publicKeyOrToken;
+            }
+
+            public string m_name;
```

Comment:
The coding conventions is to just use `_` for fields. `m_` is used only used in old code that was not cleaned up to be compliant with the coding conventions.

Reply:
Uh, I thought `_xx` is old or is used for serialization compat purposes while `m_xxx/s_xxx` is the new thing